sirepo/package_data/static/js/radia.js

```diff
@@ -3723,89 +3780,25 @@ SIREPO.viewLogic('objectShapeView', function(appState, panelState, radiaService,
 
     $scope.whenSelected = function() {
         modelType = appState.models.geomObject.type;
+        $scope.modelData = appState.models[$scope.modelName];
+        radiaService.updateModelAndSuperClasses(modelType, $scope.modelData);
```

Comment:
I don't fully understand the super class stuff yet so excuse my ignorance but is there a reason the state has to be denormalized across the model and all super classes? I ask because there is something off to me about having to call this method.

Reply:
Also, does a saveChanges have to happen after?

Reply:
The modelField directive needs the model and field, so if we have
```
baseModel: {
   baseField: ["Base Field", "BaseType", "Base Default"]
},
derivedModel: {
    "_super": ["_", "model", "baseModel"],
    "derivedField": ["Derived Field", "DerivedType", "Derived Default"]
}
```
and we want to edit ```baseField``` on the ```derivedModel```, we have to supply ```baseModel``` to modelField. We should be editing the object of interest  instead of its ancestors, but this gets the job done for now.

Also, in this case we can't save because it will instantly close a dialog. The user has not modified anything yet.